nrf_802154_sl/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+# Calculates relative path to the library to be uses taking into account
+# build type of the library, SoC type, floating point ABI, supported FEM type.
+# Note: Subdirectory layout is specific to nrf_802154_sl lib
+function(calculate_lib_path lib_path)
+  # Select build type of the library to link with
+  set(build_type_dir "Release")
+
+  # Select SoC
+  if(CONFIG_SOC_NRF52840)
+    set(soc_dir "nrf52840_xxAA")
+  elseif(CONFIG_SOC_NRF52833)
+    set(soc_dir "nrf52833_xxAA")
+  elseif(CONFIG_SOC_NRF52820)
+    set(soc_dir "nrf52820_xxAA")
+  elseif(CONFIG_SOC_NRF52811)
+    set(soc_dir "nrf52811_xxAA")
+  else()
+    message(WARNING "Selected SoC is not supported by the nrf_802154_sl lib.")
```

Comment:
Nothing wrong as an additional safeguard, but I hope the depends in Kconfig are written correctly, so that this situation cannot arise when using `menuconfig`